cmd/configschema/configschema/comments.go

```diff
@@ -23,14 +23,17 @@ import (
 )
 
 // commentsForStruct returns a map of fieldname -> comment for a struct
-func commentsForStruct(v reflect.Value, dr DirResolver) map[string]string {
+func commentsForStruct(v reflect.Value, dr DirResolver) (map[string]string, error) {
 	elem := v
 	if v.Kind() == reflect.Ptr {
 		elem = v.Elem()
 	}
-	dir := dr.PackageDir(elem.Type())
+	dir, err := dr.PackageDir(elem.Type())
+	if err != nil {
+		return nil, err
+	}
 	name := trimPackage(elem)
-	return commentsForStructName(dir, name)
+	return commentsForStructName(dir, name), nil
```

Comment:
Not required here, but I wonder if it's better overall to only panic in main and use errors in all helpers.  This would allow for expanded unit tests to vet error conditions, though `assert.Panics()` could be used in the meantime.

Reply:
So that would be to basically return the errors all the way up to the external caller?

Reply:
sry not sure how I missed notifications for this.  Right so the caller makes the decision to panic like you did and it doesn't limit the ability to use a given function in ~harmless contexts.  In cases like unsafe or reflection usage where there is no sensible path forward/back panicking makes more sense.